From 7dec673ea949fd7278993b76e33a05f561294e43 Mon Sep 17 00:00:00 2001
From: Benjamin Young <benjaminzyoung@gmail.com>
Date: Fri, 13 Oct 2017 14:29:20 -0700
Subject: [PATCH 043/105] cmd: fastboot: Enable FASTBOOT_FLASH_NAND for
 SUNXI_NAND devices

Encountered an issue where fastboot can't write to NAND on a CHIP_pro,
the symbol was neither present in the board's config header, nor the
Kconfig, this patch puts it in the Kconfig and defaults on when
SUNXI_NAND is selected.

Signed-off-by: Ben Young <computermouth@crunchbangplusplus.org>
Signed-off-by: Maxime Ripard <maxime.ripard@free-electrons.com>
---
 cmd/fastboot/Kconfig | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/cmd/fastboot/Kconfig b/cmd/fastboot/Kconfig
index 214bbc23fc..4ce7a775e2 100644
--- a/cmd/fastboot/Kconfig
+++ b/cmd/fastboot/Kconfig
@@ -81,6 +81,16 @@ config FASTBOOT_FLASH_MMC_DEV
 	  regarding the non-volatile storage device. Define this to
 	  the eMMC device that fastboot should use to store the image.
 
+config FASTBOOT_FLASH_NAND_DEV
+	int "Define FASTBOOT NAND FLASH default device"
+	depends on FASTBOOT_FLASH && NAND
+	depends on CMD_MTDPARTS
+	default 0 if ARCH_SUNXI && NAND_SUNXI
+	help
+	  The fastboot "flash" command requires additional information
+	  regarding the non-volatile storage device. Define this to
+	  the NAND device that fastboot should use to store the image.
+
 config FASTBOOT_GPT_NAME
 	string "Target name for updating GPT"
 	depends on FASTBOOT_FLASH
-- 
2.13.6

